runtime/src/message_processor.rs

```diff
@@ -263,6 +263,7 @@ pub struct ThisInvokeContext<'a> {
     rent: Rent,
     pre_accounts: Vec<PreAccount>,
     account_deps: &'a [(Pubkey, RefCell<Account>)],
+    keyed_accounts: &'a [KeyedAccount<'a>],
```

Comment:
Can we instead keep the keyed_accounts stored with the program id for each call stack level:
```rust
call_stack: Vec<(Pubkey, &'a [KeyedAccount<'a>)>,
```
And `push` and `pop` just push and pop entries and `get_keyed_accounts` returns the keyed accounts from the top?

Reply:
I tried that but it does not rid us of all the unsafe statements here,
because there still needs to be a lifetime upcast from the lifetime of the freshly created `keyed_accounts`,
which are only valid in this stack frame, to the general lifetime of the `invoke_context`.

However, if we would not create "new" `keyed_accounts` here and only reorder existing ones, then the problem should disappear entirely.